Orange/data/table.py

```diff
@@ -769,246 +660,152 @@ def __repr__(self):
         return s
 
     def clear(self):
-        """Remove all rows from the table."""
-        if not self._check_all_dense():
-            raise ValueError("Tables with sparse data cannot be cleared")
-        del self[...]
+        """Remove all rows from the table in-place."""
+        self.drop(self.index, inplace=True)
 
-    def append(self, instance):
+    def append(self, row):
         """
-        Append a data instance to the table.
-
-        :param instance: a data instance
-        :type instance: Orange.data.Instance or a sequence of values
+        Append a new row to the table.
+        row can either be a single value (broadcast),
+        a list-like of values or a TableSeries (a single row slice).
         """
-        self.insert(len(self), instance)
-
-    def insert(self, row, instance):
+        new_ix = Table.new_id()
+        self.loc[new_ix] = row
+
+    # TODO: this is awkward; inserting (manually, by the user) via .loc[]
+    #       doesn't maintain proper indexing, so this has to be used
+    # TODO: this inserts at the end, a proxy for append
+    # TODO: this hides df.insert(), but that is for columns (not rows)
+    @deprecated('Table.append()')
+    def insert(self, spot, instance):
```

Comment:
I'd just override insert and handle everything in __setitem__. Insert in pandas (horizontal) is different than in legacy Orange (vertical) anyway.